bootstrap/updater.go

```diff
@@ -419,6 +420,33 @@ func validateData(cfg Config, update *Update) (*VerifiedUpdate, error) {
 	return verified, nil
 }
 
+func renameLegacyFile(fs afero.Fs, path string) string {
+	var idx int
+	for _, suffix := range []string{SuffixBoostrap, SuffixBeacon, SuffixActiveSet} {
+		if strings.HasSuffix(path, suffix) {
+			return path
+		}
+	}
+	for _, suffix := range []string{SuffixBoostrap, SuffixBeacon, SuffixActiveSet} {
+		idx = strings.Index(path, fmt.Sprintf("-%s-", suffix))
+		if idx > -1 {
+			break
+		}
+	}
+	if idx < 0 {
+		return ""
+	}
+	newPath := path[:idx+suffixLen+1]
+	if exists, _ := afero.Exists(fs, newPath); exists {
+		_ = fs.Remove(path)
```

Comment:
Could this cause issues? Removing an existing file to make space for the old file to be renamed with new name?

Reply:
if you start a node repeatedly, you will see the following. 
```
$ ls -l data/bootstrap/2
total 1404
-r-------- 1 k k 158713 Aug 15 22:12 epoch-2-update-bs-2023-08-15T20-12-53
-r-------- 1 k k 158713 Aug 15 22:53 epoch-2-update-bs-2023-08-15T20-53-20
-r-------- 1 k k 158713 Aug 15 23:23 epoch-2-update-bs-2023-08-15T21-23-44
-r-------- 1 k k 158713 Aug 16 02:39 epoch-2-update-bs-2023-08-16T00-39-21
-r-------- 1 k k 158713 Aug 16 02:47 epoch-2-update-bs-2023-08-16T00-47-43
-r-------- 1 k k 158713 Aug 16 03:01 epoch-2-update-bs-2023-08-16T01-01-58
-r-------- 1 k k 158713 Aug 16 19:26 epoch-2-update-bs-2023-08-16T17-26-07
-r-------- 1 k k 158713 Aug 16 20:15 epoch-2-update-bs-2023-08-16T18-15-46
-r-------- 1 k k 158713 Aug 16 20:16 epoch-2-update-bs-2023-08-16T18-16-00
```
those files are all identical. we just need one, and `newPath` already exists (because it renamed the first one to `newPath` on line 444. so it's safe to delete here.